src/completions.jl

```diff
@@ -99,6 +94,100 @@ function basecompletionadapter(
   return comps
 end
 
+# Fuzzy completion
+# ----------------
+
+handle("fuzzycompletionadapter") do data
+  @destruct [
+    # general
+    line,
+    mod || "Main",
+    path || nothing,
+    # local context
+    context || "",
+    row || 1,
+    startRow || 0,
+    column || 1,
+    # configurations
+    force || false
+  ] = data
+
+  return fuzzycompletionadapter(
+    # general
+    line, mod,
+    # local context
+    context, row - startRow, column,
+    # configurations
+    force
+  )
+end
+
+function fuzzycompletionadapter(
```

Comment:
This seems fairly similar to the `replcompletionadapter` -- maybe we could have an `is_fuzzy` kwarg and do something like
```
repl_provider = is_fuzzy ? FuzzyCompletions : REPLCompletions
repl_provider.completions(...)
```
plus have an `if` block for the different filter methods below?

Reply:
hm, then the code after `repl_provider` used would be really type-instable (I confirmed that Julia 1.4 can't infer the type of `repl_provider.completions`, for example)

I haven't taken any actual benchmark, but I would want to endure the verbose code for performance here.

Reply:
Hm fair enough then. I don't think the dynamic dispatch would make much of a difference, but let's just keep this as-is then.

Reply:
> I don't think the dynamic dispatch would make much of a difference

yeah maybe. I may be too nervous about dynamic dispatches..
I will refactor them in the future when I get confidence that dynamic dispatches here don't matter.